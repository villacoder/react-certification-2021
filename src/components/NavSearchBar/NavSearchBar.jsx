import React, { useState, useContext } from 'react';
import { Navbar, Nav, Form, FormControl, Button } from 'react-bootstrap';
import Switch from '@material-ui/core/Switch';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { Link } from 'react-router-dom';
import { BiSearchAlt2 } from 'react-icons/bi';
import { FaRegUserCircle } from 'react-icons/fa';
import { AppContext } from '../../providers/App/AppProvider';
import { types } from '../../providers/App/types';

export const NavSearchBar = () => {
  const { dispatch } = useContext(AppContext);
  const [searchTerm, setSearchTerm] = useState('');
  const [checked, setChecked] = React.useState(false);

  const toggleChecked = () => {
    setChecked((prev) => !prev);
    dispatch({
      type: types.setDarkMode,
    });
  };

  const onSubmit = (event) => {
    dispatch({
      type: types.setPlayVideo,
      payload: false,
    });
    dispatch({
      type: types.setSearchButton,
      payload: true,
    });
    dispatch({
      type: types.setPassToChild,
      payload: searchTerm,
    });
    event.preventDefault();
  };

  return (
    <Navbar style={{ backgroundColor: '#1C5476' }} expand="lg">
      <Navbar.Brand
        style={{
          color: '#FFFFFF',
          padding: '10px 10px',
          justifyContent: 'space-around',
        }}
        href="/"
      >
        WizeTube
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="navbarScroll" />
      <Navbar.Collapse id="navbarScroll" style={{ justifyContent: 'space-around' }}>
        <Nav
          className="mr-auto my-2 my-lg-0"
          style={{
            maxHeight: '100px',
            backgroundColor: '#1C5476',
            justifyContent: 'space-around',
          }}
          navbarScroll
        >
          {' '}
        </Nav>
        <div style={{ justifyContent: 'space-around' }}>
          <Form
            className="d-flex"
            onSubmit={onSubmit}
            style={{
              backgroundColor: '#3E6D8A',
              border: 'none',
              overflow: 'hidden',
              borderRadius: '5px',
              outline: 'none',
            }}
          >
            <Button
              style={{
                backgroundColor: '#3E6D8A',
                border: 'none',
                overflow: 'hidden',
                outline: 'none',
              }}
              onClick={() => {
                dispatch({
                  type: types.setSearchButton,
                  payload: true,
                });
                dispatch({
                  type: types.setPlayVideo,
                  payload: false,
                });
                dispatch({
                  type: types.setPassToChild,
                  payload: searchTerm,
                });
              }}
            >
              <BiSearchAlt2 color="white" size="30px" />
            </Button>
            <FormControl
              type="search"
              placeholder="Search.."
              aria-label="Search"
              style={{
                backgroundColor: '#3E6D8A',
                border: 'none',
                color: 'white',
                outline: 'none',
                boxShadow: 'none',
              }}
              onChange={(event) => setSearchTerm(event.target.value)}
            />
          </Form>{' '}
        </div>
        <FormControlLabel
          control={<Switch checked={checked} onChange={toggleChecked} />}
          label="Dark mode"
          style={{
            color: 'white',
          }}
        />
        <div style={{ justifyContent: 'space-around' }}>
          <Nav>
            <Link to="/login">
              {' '}
              <FaRegUserCircle color="white" size="35px" />
            </Link>
          </Nav>
        </div>
      </Navbar.Collapse>
    </Navbar>
  );
};
