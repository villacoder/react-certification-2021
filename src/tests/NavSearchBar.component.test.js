/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render } from '@testing-library/react';
import { NavSearchBar } from '../components/NavSearchBar/NavSearchBar';

describe('NavSearchBar', () => {
  test('Should render properly', () => {
    const component = render(<NavSearchBar />);
    expect(component).toMatchSnapshot();
  });
});
