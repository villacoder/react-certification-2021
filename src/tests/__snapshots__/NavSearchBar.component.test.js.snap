// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavSearchBar Should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-light"
        style="background-color: rgb(28, 84, 118);"
      >
        <a
          class="navbar-brand"
          href="/"
          style="color: rgb(255, 255, 255); padding: 10px 10px;"
        >
          WizeTube
        </a>
        <button
          aria-controls="navbarScroll"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="navbarScroll"
        >
          <div
            class="mr-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
            style="max-height: 100px; background-color: rgb(28, 84, 118);"
          >
            <div
              class="nav-item dropdown"
              style="color: rgb(255, 255, 255);"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle nav-link"
                href="#"
                id="navbarScrollingDropdown"
                role="button"
              >
                Menu
              </a>
            </div>
          </div>
          <form
            class="d-flex"
            style="background-color: rgb(62, 109, 138); overflow: hidden; border-radius: 5px; outline: none;"
          >
            <button
              class="btn btn-primary"
              style="background-color: rgb(62, 109, 138); overflow: hidden; outline: none;"
              type="button"
            >
              <svg
                color="white"
                fill="currentColor"
                height="30px"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 24 24"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.023,16.977c-0.513-0.488-1.004-0.997-1.367-1.384c-0.372-0.378-0.596-0.653-0.596-0.653l-2.8-1.337 C15.34,12.37,16,10.763,16,9c0-3.859-3.14-7-7-7S2,5.141,2,9s3.14,7,7,7c1.763,0,3.37-0.66,4.603-1.739l1.337,2.8 c0,0,0.275,0.224,0.653,0.596c0.387,0.363,0.896,0.854,1.384,1.367c0.494,0.506,0.988,1.012,1.358,1.392 c0.362,0.388,0.604,0.646,0.604,0.646l2.121-2.121c0,0-0.258-0.242-0.646-0.604C20.035,17.965,19.529,17.471,19.023,16.977z M9,14 c-2.757,0-5-2.243-5-5s2.243-5,5-5s5,2.243,5,5S11.757,14,9,14z"
                />
              </svg>
            </button>
            <input
              aria-label="Search"
              class="form-control"
              placeholder="Search.."
              style="background-color: rgb(62, 109, 138); color: white; outline: none; box-shadow: none;"
              type="search"
              value="wizeline"
            />
          </form>
          <a
            class="nav-link"
            href="/login"
          >
             
            <svg
              color="white"
              fill="currentColor"
              height="35px"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 496 512"
              width="35px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
              />
            </svg>
             
          </a>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-light"
      style="background-color: rgb(28, 84, 118);"
    >
      <a
        class="navbar-brand"
        href="/"
        style="color: rgb(255, 255, 255); padding: 10px 10px;"
      >
        WizeTube
      </a>
      <button
        aria-controls="navbarScroll"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="navbarScroll"
      >
        <div
          class="mr-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
          style="max-height: 100px; background-color: rgb(28, 84, 118);"
        >
          <div
            class="nav-item dropdown"
            style="color: rgb(255, 255, 255);"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              class="dropdown-toggle nav-link"
              href="#"
              id="navbarScrollingDropdown"
              role="button"
            >
              Menu
            </a>
          </div>
        </div>
        <form
          class="d-flex"
          style="background-color: rgb(62, 109, 138); overflow: hidden; border-radius: 5px; outline: none;"
        >
          <button
            class="btn btn-primary"
            style="background-color: rgb(62, 109, 138); overflow: hidden; outline: none;"
            type="button"
          >
            <svg
              color="white"
              fill="currentColor"
              height="30px"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 24 24"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.023,16.977c-0.513-0.488-1.004-0.997-1.367-1.384c-0.372-0.378-0.596-0.653-0.596-0.653l-2.8-1.337 C15.34,12.37,16,10.763,16,9c0-3.859-3.14-7-7-7S2,5.141,2,9s3.14,7,7,7c1.763,0,3.37-0.66,4.603-1.739l1.337,2.8 c0,0,0.275,0.224,0.653,0.596c0.387,0.363,0.896,0.854,1.384,1.367c0.494,0.506,0.988,1.012,1.358,1.392 c0.362,0.388,0.604,0.646,0.604,0.646l2.121-2.121c0,0-0.258-0.242-0.646-0.604C20.035,17.965,19.529,17.471,19.023,16.977z M9,14 c-2.757,0-5-2.243-5-5s2.243-5,5-5s5,2.243,5,5S11.757,14,9,14z"
              />
            </svg>
          </button>
          <input
            aria-label="Search"
            class="form-control"
            placeholder="Search.."
            style="background-color: rgb(62, 109, 138); color: white; outline: none; box-shadow: none;"
            type="search"
            value="wizeline"
          />
        </form>
        <a
          class="nav-link"
          href="/login"
        >
           
          <svg
            color="white"
            fill="currentColor"
            height="35px"
            stroke="currentColor"
            stroke-width="0"
            style="color: white;"
            viewBox="0 0 496 512"
            width="35px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
            />
          </svg>
           
        </a>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
